modVM/src/org/aion/vm/TransactionResult.java

```diff
@@ -81,6 +83,23 @@ public TransactionResult() {
         super(ResultCode.SUCCESS, 0);
     }
 
+    /**
+     * Returns a big-endian binary encoding of this TransactionResult.
+     *
+     * @return a big-endian binary encoding of this TransactionResult.
+     */
+    public final byte[] toBytes() {
+        ByteBuffer buffer = ByteBuffer.allocate(4 + 8 + 4 + (output == null ? 0 : output.length));
+        buffer.order(ByteOrder.BIG_ENDIAN);
+        buffer.putInt(code.toInt());
+        buffer.putLong(nrgLeft);
+        buffer.putInt(output == null ? 0 : output.length);
+        if (output != null) {
```

Comment:
The behavior is different compared with the ExecutionResult.toBytes(), any reason? if the behavior can be the same, can we put this method to the abstract class?

Reply:
Yes the two are the same. At first it seemed inappropriate to me to produce sub-class encodings in a superclass, since the children may acquire fields or other changes over time that would demand their own encodings. I suppose though the better choice is to have a generic toBytes in the super and then override when necessary, so I'll put that back as it was